src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/ReadGrainMapper3D.cpp

```diff
@@ -125,15 +125,33 @@ Result<> ReadGrainMapper3D::copyDctData(GrainMapperReader& reader, hid_t fileId)
   reader.findAvailableDctDatasets(labDctGid);
   auto dctDataSets = reader.getDctDatasetNames();
 
-  std::vector<std::string> floatDataSets = {GM3DConst::k_CompletenessName, GM3DConst::k_EulerZXZName, GM3DConst::k_EulerZYZName, GM3DConst::k_QuaternionName, GM3DConst::k_RodriguesName};
-  std::vector<std::string> in32DataSets = {GM3DConst::k_GrainIdName};
-  std::vector<std::string> uint8DataSets = {GM3DConst::k_MaskName, GM3DConst::k_IPF001Name, GM3DConst::k_IPF010Name, GM3DConst::k_IPF100Name, GM3DConst::k_PhaseIdName};
+  std::set<std::string> floatDataSets;
+  std::set<std::string> in32DataSets;
+  std::set<std::string> uint8DataSets;
+
+  auto nameToDataTypeMap = reader.getNameToDataTypeMap();
+  for(const auto& entry : nameToDataTypeMap)
+  {
+    if(entry.second == DataType::uint8)
+    {
+      uint8DataSets.insert(entry.first);
+    }
+    if(entry.second == DataType::int32)
+    {
+      in32DataSets.insert(entry.first);
+    }
+    if(entry.second == DataType::float32)
+    {
+      floatDataSets.insert(entry.first);
+    }
```

Comment:
```suggestion
    if(entry.second == DataType::float32)
    {
      floatDataSets.insert(entry.first);
    }
    else if(entry.second == DataType::uint8)
    {
      uint8DataSets.insert(entry.first);
    }
    else if(entry.second == DataType::int32)
    {
      in32DataSets.insert(entry.first);
    }
```
Order by most common and convert to if/else for lazy evaluation